dragonfly/item/potion/potion.go

```diff
@@ -0,0 +1,256 @@
+package potion
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/entity/effect"
+	"time"
+)
+
+// Potion holds the effects given by the potion type.
+type Potion struct {
+	potion
+	Effects []effect.Effect
+}
+
+// Water ...
+func Water() Potion {
+	return Potion{}
+}
+
+// Mundane ...
+func Mundane() Potion {
+	return Potion{potion: 1}
+}
+
+// LongMundane ...
+func LongMundane() Potion {
+	return Potion{potion: 2}
+}
+
+// Thick ...
+func Thick() Potion {
+	return Potion{potion: 3}
+}
+
+// Awkward ...
+func Awkward() Potion {
+	return Potion{potion: 4}
+}
+
+// NightVision ...
+func NightVision() Potion {
+	return Potion{5, []effect.Effect{effect.NightVision{}.WithSettings(time.Duration(3)*time.Minute, 1, false)}}
+}
+
+// LongNightVision ...
+func LongNightVision() Potion {
+	return Potion{6, []effect.Effect{effect.NightVision{}.WithSettings(time.Duration(8)*time.Minute, 1, false)}}
+}
+
+// Invisibility ...
+func Invisibility() Potion {
+	return Potion{7, []effect.Effect{effect.Invisibility{}.WithSettings(time.Duration(3)*time.Minute, 1, false)}}
+}
+
+// LongInvisibility ...
+func LongInvisibility() Potion {
+	return Potion{8, []effect.Effect{effect.Invisibility{}.WithSettings(time.Duration(8)*time.Minute, 1, false)}}
+}
+
+// Leaping ...
+func Leaping() Potion {
+	return Potion{9, []effect.Effect{effect.JumpBoost{}.WithSettings(time.Duration(3)*time.Minute, 1, false)}}
+}
+
+// LongLeaping ...
+func LongLeaping() Potion {
+	return Potion{10, []effect.Effect{effect.JumpBoost{}.WithSettings(time.Duration(8)*time.Minute, 1, false)}}
+}
+
+// StrongLeaping ...
+func StrongLeaping() Potion {
+	return Potion{11, []effect.Effect{effect.JumpBoost{}.WithSettings(time.Duration(90)*time.Second, 2, false)}}
+}
+
+// FireResistance ...
+func FireResistance() Potion {
+	return Potion{potion: 12} //TODO
+}
+
+// LongFireResistance ...
+func LongFireResistance() Potion {
+	return Potion{potion: 13} //TODO
```

Comment:
Could you please add something like `Implement fire resistance` at these TODOs so we can easily find them once looking for fire resistance TODOs?